lib/sisimai/bite/email/ezweb.rb

```diff
@@ -0,0 +1,260 @@
+            # mail server.
+            #
+            #    Recipient: <******@ezweb.ne.jp>
+            #    >>> RCPT TO:<******@ezweb.ne.jp>
+            #    <<< 550 <******@ezweb.ne.jp>: User unknown
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[<]([^ ]+[@][^ ]+)[>]\z/) ||
+                    e.match(/\A[<]([^ ]+[@][^ ]+)[>]:?(.*)\z/) ||
+                    e.match(/\A[ \t]+Recipient: [<]([^ ]+[@][^ ]+)[>]/)
+
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                push dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+
+              r = Sisimai::Address.s3s4(cv[1])
+              if Sisimai::RFC5322.is_emailaddress(r)
+                v['recipient'] = r
+                recipients += 1
+              end
+
+            elsif cv = e.match(/\A[Ss]tatus:[ ]*(\d[.]\d+[.]\d+)/)
+              # Status: 5.1.1
+              # Status:5.2.0
+              # Status: 5.1.0 (permanent failure)
+              v['status'] = cv[1]
+
+            elsif cv = e.match(/\A[Aa]ction:[ ]*(.+)\z/)
+              # Action: failed
+              v['action'] = cv[1].downcase
+
+            elsif cv = e.match(/\A[Rr]emote-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+              # Remote-MTA: DNS; mx.example.jp
+              v['rhost'] = cv[1].downcase
+
+            elsif cv = e.match(/\A[Ll]ast-[Aa]ttempt-[Dd]ate:[ ]*(.+)\z/)
+              # Last-Attempt-Date: Fri, 14 Feb 2014 12:30:08 -0500
+              v['date'] = cv[1]
+
+            else
+              next if Sisimai::String.is_8bit(e)
+              if cv = e.match(/\A[ \t]+[>]{3}[ \t]+([A-Z]{4})/)
+                #    >>> RCPT TO:<******@ezweb.ne.jp>
+                v['command'] = cv[1]
+
+              else
+                # Check error message
+                if rxmessages.find { |a| e =~ a }
+                  # Check with regular expressions of each error
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + e
+
+                else
+                  # >>> 550
+                  v['alterrors'] ||= ''
+                  v['alterrors']  += ' ' + e
+                end
+              end
+            end
+          end
+        end
+        return nil if recipients.zero?
+
+        dscontents.map do |e|
+          if e['alterrors'] && e['alterrors'].size > 0
+            # Copy alternative error message
+            e['diagnosis'] ||= e['alterrors']
+            if e['diagnosis'] =~ /\A[-]+/ || e['diagnosis'] =~ /__\z/
+              # Override the value of diagnostic code message
+              e['diagnosis'] = e['alterrors'] if e['alterrors'].size > 0
+            end
+            e.delete('alterrors')
+          end
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          if mhead['x-spasign'] && mhead['x-spasign'] == 'NG'
+            # Content-Type: text/plain; ..., X-SPASIGN: NG (spamghetti, au by EZweb)
```

Comment:
__[rubocop]__
* Performance/EndWith: Use `String#end_with?` instead of a regex match anchored to the end of the string. [:ref](https://github.com/JuanitoFatas/fast-ruby#stringmatch-vs-stringstart_withstringend_with-code-start-code-end)

<!-- Comment by [SideCI](https://sideci.com) -->